compiler/hash-parser/src/gen.rs

```diff
@@ -133,13 +131,20 @@ where
         }
     }
 
+    /// Function that skips the next token without explicitly looking up the
+    /// token in the stream and avoiding the additional computation.
+    #[inline(always)]
+    pub(crate) fn skip_token(&self) {
+        self.offset.update(|x| x + 1);
+    }
+
     /// Function that increases the offset of the next token
     pub(crate) fn next_token(&self) -> Option<&Token<'c>> {
         let value = self.stream.get(self.offset.get());
 
         if value.is_some() {
-            // @@Dumbness: because Cell::update is unsafe
-            self.offset.set(self.offset.get() + 1);
+            // @@UnsafeLibUsage
+            self.offset.update::<_>(|x| x + 1);
```

Comment:
Is this really faster than the way it was before?  Surely not, right?

Reply:
This is just more ergonomic than getting, then setting.